shared/chat/conversation/input-area/normal/platform-input.desktop.js

```diff
@@ -215,143 +210,154 @@ class PlatformInput extends Component<PlatformInputProps & OverlayParentProps, S
+                <Kb.Text type="BodyTinyBold" style={styles.time}>
+                  {formatDurationShort(this.props.explodingModeSeconds * 1000)}
+                </Kb.Text>
+              ) : (
+                <Kb.Icon
+                  className="timer"
+                  onClick={this.props.isEditing ? undefined : this._toggleShowingMenu}
+                  style={Kb.iconCastPlatformStyles(styles.timerIcon)}
+                  type="iconfont-timer"
                 />
               )}
-            {flags.explodingMessagesEnabled && (
-              <HoverBox
-                onClick={this._toggleShowingMenu}
-                ref={this.props.setAttachmentRef}
-                style={collapseStyles([
-                  styles.explodingIconContainer,
-                  !!(this.props.isExplodingNew || this.props.explodingModeSeconds) && {
-                    marginRight: globalMargins.tiny,
-                  },
-                ])}
-              >
-                <Icon
-                  className="bomb"
-                  color={this.props.explodingModeSeconds === 0 ? null : globalColors.black_75}
-                  onClick={this._toggleShowingMenu}
-                  style={styleIcon}
-                  type="iconfont-bomb"
-                />
-                <ExplodingMeta
-                  explodingModeSeconds={this.props.explodingModeSeconds}
-                  isNew={this.props.isExplodingNew}
-                />
-              </HoverBox>
+            </HoverBox>
+          )}
+          {this.props.isEditing && (
+            <Kb.Box onClick={this.props.onCancelEditing} style={styles.cancelEditing}>
+              <Kb.Text style={styles.cancelEditingText} type="BodySmallSemibold">
+                Cancel
+              </Kb.Text>
+            </Kb.Box>
+          )}
+          <input
+            type="file"
+            style={styles.hidden}
+            ref={this._filePickerSetRef}
+            onChange={this._pickFile}
+            multiple={true}
+          />
+          <Kb.Input
+            className={'mousetrap' /* className needed so key handler doesn't ignore hotkeys */}
+            autoFocus={false}
+            small={true}
+            style={collapseStyles([
+              styles.input,
+              {
+                backgroundColor: this.props.isEditing ? globalColors.yellow3 : globalColors.white,
+              },
+            ])}
+            ref={this._inputSetRef}
+            hintText={hintText}
+            hideUnderline={true}
+            onChangeText={this._onChangeText}
+            uncontrolled={true}
+            multiline={true}
+            rowsMin={1}
+            rowsMax={10}
+            onKeyDown={this._onKeyDown}
+            onEnterKeyDown={this._onEnterKeyDown}
+          />
+          {flags.explodingMessagesEnabled &&
+            this.props.isExploding &&
+            !this.props.isEditing &&
+            !this.state.hasText && (
+              <Kb.Icon
```

Comment:
this is the boom placeholder icon? :sheep: might be nice to add a comment

Reply:
Done!